server.go

```diff
@@ -156,6 +180,32 @@ func (e *errPeerAlreadyConnected) Error() string {
 	return fmt.Sprintf("already connected to peer: %v", e.peer)
 }
 
+// peerAccessStatus denotes the p2p access status of a given peer. This will be
+// used to assign peer ban scores that determine an action the server will
+// take.
+type peerAccessStatus int
+
+const (
+	// restricted indicates that the peer only has access to the limited
+	// number of "free" reserved slots.
+	restricted peerAccessStatus = iota
```

Comment:
I'd suggest to use naming with a prefix as these words are just too general. Maybe `peerStatusTemporary` etc.

Reply:
done